### PR TITLE
refactor!: remove deprecations

### PR DESCRIPTION
**Pull request by nsbarsukov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [X] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## Does this PR introduce a breaking change?

- [X] Yes
- [ ] No

### CODE REVIEW

File: projects/core/components/data-list/option/option.component.ts [hunk 1]
```diff
@@ -61,7 +61,6 @@ export class TuiOptionComponent<T = unknown> implements OnDestroy {
     @Input()
     value?: T;
 
-    // TODO: 3.0 Fix dataList type when updated to Ivy and compilation drops metadata
```
Comment (nsbarsukov) [inline projects/core/components/data-list/option/option.component.ts:64]: Already fixed in [the current PR](https://github.com/Tinkoff/taiga-ui/pull/1816/files#diff-d6a2eabb7cb5a90bd778bb2b977c8962ae50b8a50d426c188e8fc43b8f8961d7R75)

File: projects/core/utils/polyfills/inner-html.ts [hunk 2]
```diff
@@ -1,49 +0,0 @@
-import {TuiXmlParsingException} from '@taiga-ui/cdk';
-
-/**
- * @deprecated
- * TODO: remove in v3.0
- */
-// eslint-disable-next-line @typescript-eslint/naming-convention
-export function innerHTML(documentRef: Document): void {
```
Comment (nsbarsukov) [inline projects/core/utils/polyfills/inner-html.ts:8]: If I've not mistaken, this polyfill was for IE, right ?

File: projects/kit/tokens/tag-status.ts [hunk 3]
```diff
@@ -1,10 +0,0 @@
-import {InjectionToken} from '@angular/core';
-import {TuiStatus} from '@taiga-ui/kit/types';
-
-// TODO: 3.0 use TUI_TAG_OPTIONS instead of it
-export const TUI_TAG_STATUS = new InjectionToken<TuiStatus>(
```
Comment (nsbarsukov) [inline projects/kit/tokens/tag-status.ts:5]: oops, I have made a mistake.

I've mistakenly moved it to `TUI_INPUT_TAG_OPTIONS` (not TUI_TAG_OPTIONS).

Comment (nsbarsukov) [reply projects/kit/tokens/tag-status.ts:5]: I'll fix it soon)

**Review by splincode — APPROVED** [review]

File: projects/kit/tokens/tag-status.ts (hunk 3 above)
Comment (nsbarsukov) [reply projects/kit/tokens/tag-status.ts:5]: fixed ✅

File: projects/core/components/data-list/option/option.component.ts (hunk 1 above)
Comment (nsbarsukov) [inline projects/core/components/data-list/option/option.component.ts:64]: Already fixed in [the current PR](https://github.com/Tinkoff/taiga-ui/pull/1816/files#diff-d6a2eabb7cb5a90bd778bb2b977c8962ae50b8a50d426c188e8fc43b8f8961d7R75)

File: projects/kit/tokens/tag-status.ts [hunk 4]
```diff
@@ -1,10 +0,0 @@
-import {InjectionToken} from '@angular/core';
-import {TuiStatus} from '@taiga-ui/kit/types';
-
-// TODO: 3.0 use TUI_TAG_OPTIONS instead of it
```
Comment (nsbarsukov) [inline projects/kit/tokens/tag-status.ts:4]: hmmm...

Current `TUI_TAG_STATUS`-token has `primary`-status (by default).
`TUI_TAG_OPTIONS` has `default`-status (by default).

🤔

**Review by waterplea — APPROVED** [review]

File: projects/core/components/button/button-options.ts [hunk 5]
```diff
@@ -4,12 +4,8 @@ import {TuiSizeXL, TuiSizeXS} from '@taiga-ui/core/types';
 
 export interface TuiButtonOptions {
     readonly size: TuiSizeXS | TuiSizeXL;
-    readonly appearance:
-        | keyof Record<TuiAppearance, string>
-        | TuiAppearance
-        | string
-        | null; // TODO: 3.0 need remove `null`
-    readonly shape: 'square' | 'rounded' | null; // TODO: 3.0 need remove `null`
+    readonly appearance: keyof Record<TuiAppearance, string> | TuiAppearance | string;
```
[the remark this answers is not in the document]
Comment (nsbarsukov) [reply projects/core/components/button/button-options.ts:7]: For autocomplete suggestions ?

suggestions

[the remark this answers is not in the document]
Comment (splincode) [reply projects/core/components/button/button-options.ts:7]: `readonly appearance: 'primary' | 'accent' | ... | string;` why not?

[the remark this answers is not in the document]
Comment (nsbarsukov) [reply projects/core/components/button/button-options.ts:7]: Should i replace all usage of `const enum TuiAppearance` with `'primary' | 'accent' | ... | ... `?

[the remark this answers is not in the document]
Comment (waterplea) [reply projects/core/components/button/button-options.ts:7]: Yes, let's drop that enum

[the remark this answers is not in the document]
Comment (nsbarsukov) [reply projects/core/components/button/button-options.ts:7]: ok, i will drop it in next PR 👌

**Pull request merged**